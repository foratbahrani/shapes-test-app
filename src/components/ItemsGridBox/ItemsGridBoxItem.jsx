import React from 'react';
import Shape from '../Shape/Shape';
import './ItemsGridBoxItem.scss';

const ItemsGridBoxItem = () => {
	return (
		<div className='itemsGridBoxItemContainer'>
			<div className='itemsGridBoxItem'>
				<div className='itemsGridBoxItem-content'>
					<Shape size='lg' color='red' shape='square' />
				</div>
			</div>
		</div>
	);
};

export default ItemsGridBoxItem;
