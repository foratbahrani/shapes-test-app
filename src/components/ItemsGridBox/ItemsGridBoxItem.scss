$this-padding: 5px;
$this-tile-size-xs: 100%;
$this-tile-size-sm: 50%;
$this-tile-size-md: 33.3333%;
$this-tile-size-lg: 25%;

.itemsGridBoxItemContainer {
	position: relative;
	display: inline-block;
	height: 0;
	width: $this-tile-size-xs;
	padding-top: $this-tile-size-xs;

	font-size: 20px;
	color: white;
	font-weight: bold;
	text-align: center;

	@media (min-width: 500px) {
		width: $this-tile-size-sm;
		padding-top: $this-tile-size-sm;
	}

	@media (min-width: 768px) {
		width: $this-tile-size-md;
		padding-top: $this-tile-size-md;
	}

	@media (min-width: 1000px) {
		width: $this-tile-size-lg;
		padding-top: $this-tile-size-lg;
	}
}
.itemsGridBoxItem {
	position: absolute;
	display: flex;
	justify-content: center;
	align-items: center;
	top: 0;
	bottom: 0;
	right: 0;
	left: 0;
	margin: 10px;
	border: #f1f1f1 solid 2px;
	border-radius: 8px;
}
