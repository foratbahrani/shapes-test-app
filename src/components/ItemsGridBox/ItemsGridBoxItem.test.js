import { shallow } from 'enzyme';
import React from 'react';
import ItemsGridBoxItem from './ItemsGridBoxItem';

it('matches snapshot', () => {
	expect(shallow(<ItemsGridBoxItem color='red' shape='rectangle' />)).toMatchSnapshot();
});
