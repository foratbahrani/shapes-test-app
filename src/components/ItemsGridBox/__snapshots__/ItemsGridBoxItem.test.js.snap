// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="itemsGridBoxItemContainer"
>
  <div
    className="itemsGridBoxItem"
  >
    <div
      className="itemsGridBoxItem-content shrink"
    >
      <Shape
        color="red"
        shape="rectangle"
        size="lg"
      />
    </div>
  </div>
</div>
`;
